### PR TITLE
feat: transactions live sync

### LINKED ISSUE #96: Reload transaction details data when it gets synced
### Please describe the problem that your request should resolve.
Right now it's necessary to manually reload the page to check if a transaction has been synced and obtain additional data.

### Describe the solution you'd like

Automatically reload transaction data once it gets synced. WS listen?

### Is it already possible to achieve the same outcome in any other way? If so, how?

Manually reload the page.

### LINKED ISSUE #446: Loading indicator of transactions waiting for mdw sync
When data from mdw is not available we display a lot of placeholders "---" but there's nothing indicating that data will be displayed once mdw is synced.

Example:

image

### PR DESCRIPTION
**Pull request by lukeromanowicz**
## Description
Resolves #446 and Resolves #96

Adds loading indicator for transactions that have not been synced yet by the middleware 

## Demo
Because the MDW was heavily optimized recently, it's almost impossible to get a not synced transaction and redirect to the details page in time. Therefor I simulated this behavior in the demo.

https://github.com/aeternity/aescan/assets/46789227/6f00a38a-c9dd-4120-a7f7-283fa3fbbb19

## Checklist:

- [x] I have read and followed the [Contributing Guide](../../CONTRIBUTING.md)  
- [x] My change does not require a change to the documentation.

### CODE REVIEW

File: src/pages/transactions/[id].vue [hunk 1]
```diff
@@ -44,12 +49,34 @@ const {
...
+
+  onBeforeRouteUpdate(
+    (_to, _from, next) => {
+      subscribedTransactionId.value = null
+      next()
+    },
+  )
+
+  onBeforeRouteLeave(
+    (_to, _from, next) => {
```
Comment (janmichek) [inline src/pages/transactions/[id].vue:74]: These params are not needed I guess

File: src/stores/webSocket.js [hunk 2]
```diff
@@ -52,11 +71,16 @@ export const useWebSocket = defineStore('webSocket', () => {
       return
     }
     const parsedData = camelcaseKeysDeep(JSON.parse(data))
-    await processSocketMessage(parsedData)
+    if (parsedData.subscription === 'Object' && parsedData.payload?.hash === subscribedTransactionId.value) {
```
Comment (janmichek) [inline src/stores/webSocket.js:74]: Could this be explained via extra const binding?

**Review by janmichek — CHANGES_REQUESTED** [review]
Looks solid! However I am not sure how to test it.
Added some minor code suggestions

File: src/stores/transactionDetails.js [hunk 3]
```diff
@@ -69,11 +67,16 @@ export const useTransactionDetailsStore = defineStore('transactionDetails', () =
     transactionTypeData.value = null
   }
 
+  function processTransactionUpdate(message) {
```
Comment (janmichek) [inline src/stores/transactionDetails.js:70]: the `process` prefix it's not really clear to me what it does or what it returns. Would it make sense to name it `updateTransactionTypeData`, or similar?

File: src/pages/transactions/[id].vue (hunk 1 above)
Comment (lukeromanowicz) [reply src/pages/transactions/[id].vue:74]: well... how else can I get to next without putting two placeholders first? :)

File: src/stores/webSocket.js (hunk 2 above)
Comment (lukeromanowicz) [reply src/stores/webSocket.js:74]: I tried my best but it was really hard to come up with the name for the first part half of that if statement

File: src/stores/transactionDetails.js (hunk 3 above)
Comment (lukeromanowicz) [reply src/stores/transactionDetails.js:70]: I named it this way to preserve consistency with the other process function but sure, I don't have any strong preferences here. Updated accordingly with your recommendation

File: src/stores/webSocket.js [hunk 4]
```diff
@@ -52,11 +71,19 @@ export const useWebSocket = defineStore('webSocket', () => {
       return
     }
     const parsedData = camelcaseKeysDeep(JSON.parse(data))
-    await processSocketMessage(parsedData)
+    const isSpecificEntityMessage = parsedData.subscription === 'Object'
```
Comment (michele-franchi) [inline src/stores/webSocket.js:74]: Could you explain this condition? Is it correct or was it supposed to be a typeof check?

**Review by michele-franchi — APPROVED** [review]
It looks good to me. At the same time it looks like mdw improved sync speed so now it's basically not happening anymore but it would be still useful in case it takes longer for some cases.

File: src/stores/webSocket.js (hunk 4 above)
Comment (lukeromanowicz) [reply src/stores/webSocket.js:74]: This is correct. You can subscribe to general entity type messages (e.g. Transactions, Microblocks...) or a specific entity e.g. transaction id. This is the second case. In this scenario the message.subscription is 'Object' and the payload is specific to the entity type you subscribed.

I struggled to figure out a good name for this so originally it was without extracting to a dedicated const.

File: src/pages/transactions/[id].vue (hunk 1 above)
Comment (janmichek) [reply src/pages/transactions/[id].vue:90]: Ok, I see now.

I quess this is not gonna work?

```suggestion
    ({next}) => {
```

But the `_` prefix could be removed, right? Or why is there underscore?

**Review by janmichek — APPROVED** [review]
Thanks for explanation. Added one more hint, but approving in advance

**Comment by lukeromanowicz** [comment]
@marc0olo we discussed that but it was hard to come up with a good design on short notice. This solution is a quick win and consistent with loading data everywhere else in the app. We'll for sure improve it once a better design is figured out and can be applied globally in the application.

**Pull request merged**